src/components/volume/volume.js

```diff
@@ -219,20 +222,31 @@ class VolumeControl extends BaseComponent {
   }
 
   /**
-   * get element cordinates
-   *
-   * @method getCoords
-   * @param {HTMLElement} el element to inspect
-   * @returns {{top: number, left: number}} object with the top and left position
-   * @memberof VolumeControl
+   * Computes & returns the volume of the player according to the user horizontal click / mouse move.
+   * @param {DOMRect} dimensions - dimensions of the horizontal volume bar
+   * @param {FakeEvent} e - click / move event
+   * @return {number} - the volume of the player. a number in the range of 0 and 1.
+   * @private
    */
-  getCoords(el: HTMLElement): {top: number, left: number} {
-    let box = el.getBoundingClientRect();
+  _getHorizontalVolume(dimensions: DOMRect, e: FakeEvent): number {
+    let barWidth = dimensions.width;
+    let left = dimensions.left;
+    let clickX = (e: any).clientX;
+    return (clickX - left) / barWidth;
+  }
 
-    return {
-      top: box.top,
-      left: box.left
-    };
+  /**
+   * Computes & returns the volume of the player according to the user vertical click / mouse move.
+   * @param {DOMRect} dimensions - dimensions of the vertical volume bar
+   * @param {FakeEvent} e - click / move event
+   * @return {number} - the volume of the player. a number in the range of 0 and 1.
+   * @private
+   */
+  _getVerticalVolume(dimensions: DOMRect, e: FakeEvent): number {
+    let barHeight = dimensions.height;
+    let top = dimensions.top;
+    let clickY = (e: any).clientY;
```

Comment:
needed?

Reply:
yes cause we use the eventManager that fires here FakeEvent